client/client_test.go

```diff
@@ -1628,6 +1629,54 @@ func testTarExporterWithSocket(t *testing.T, sb integration.Sandbox) {
 	require.NoError(t, err)
 }
 
+// moby/buildkit#1418
+func testTarExporterSymlink(t *testing.T, sb integration.Sandbox) {
+	requiresLinux(t)
+	c, err := New(context.TODO(), sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	busybox := llb.Image("busybox:latest")
+	st := llb.Scratch()
+
+	run := func(cmd string) {
+		st = busybox.Run(llb.Shlex(cmd), llb.Dir("/wd")).AddMount("/wd", st)
+	}
+
+	run(`sh -c "echo -n first > foo;ln -s foo bar"`)
+
+	def, err := st.Marshal()
+	require.NoError(t, err)
+
+	destDir, err := ioutil.TempDir("", "buildkit")
+	require.NoError(t, err)
+	defer os.RemoveAll(destDir)
+
+	out := filepath.Join(destDir, "out.tar")
+	outW, err := os.Create(out)
+	require.NoError(t, err)
+	_, err = c.Solve(context.TODO(), def, SolveOpt{
+		Exports: []ExportEntry{
+			{
+				Type:  ExporterTar,
+				Attrs: map[string]string{},
+				Output: fixedWriteCloser(outW),
```

Comment:
nit: as we don't actually need a file/destDir in here using a buffer might be simpler https://github.com/moby/buildkit/blob/d13c5afeda0eb336a5b11387ca13d89689606ddc/frontend/dockerfile/dockerfile_test.go#L406-L407